### ISSUE TITLE
Suggestion for speed improvements in search.php

### ISSUE THREAD

**Issue by musicpanda** [thirtybees/thirtybees#1349]
I had a look at search.php and found a few issues that could be optimized. 

On line 227 of Search.php you find:
`        $eligibleProducts = [];
        foreach ($results as $row) {
            $eligibleProducts[] = $row['id_product'];
        }

        $eligibleProducts2 = [];
        foreach ($intersectArray as $query) {
            foreach ($db->executeS($query, true, false) as $row) {
                $eligibleProducts2[] = $row['id_product'];
            }
        }
        $eligibleProducts = array_unique(array_intersect($eligibleProducts, array_unique($eligibleProducts2)));`

What happens here is that you first make an array of "eligible products". That means that in a shop of 10,000 products that are active, available and searchable you will get an array with 10,000 product id's. 

Next you search for the keywords and you make an intersect of that result with the "eligible products". In my opinion those two queries might better be merged into one.

**Comment by getdatakick** [comment]
Yes, this seems like a good improvement. Maybe even resolve `$eligibleProducts2` first, and include them in the query for `$result`

**Comment by musicpanda** [comment]
I had been thinking about that option too. But I doubt it would be efficient. When you get a few hundred results from the tb_search* tables you would check them for availability in a query that ended with 
   WHERE id_product IN (2,4,6,7, .... ) 
and I doubt that that is very efficient.

**Comment by musicpanda** [comment]
An alternative is to include the availability check etc in the query at around line 300 where the data for the display are retrieved. That query already includes such a "WHERE id_product IN (2,4,6,7, .... )" clause.

**Comment by Zengraph** [comment]
Is there any update for this enhancement ?
